packages/base/src/features/F6Navigation.js

```diff
@@ -86,26 +82,26 @@ class F6Navigation {
 	}
 
 	updateGroups() {
-		this.setSelectedGroup(document.activeElement);
+		this.setSelectedGroup(document);
 		this.groups = getFastNavigationGroups(document.body);
 	}
 
 	setSelectedGroup(element) {
 		element = this.deepActive(element);
-
-		while (element && element.getAttribute("data-sap-ui-fastnavgroup") !== "true" && element !== document.querySelector("html")) {
+		htmlElement = document.querySelector("html");
```

Comment:
Since htmlElement will not be overridden we can define it as constant. Please move the definition of the variable at the top of the function and add new line before while loop.

Reply:
updated